mlir/lib/Dialect/Rock/Transforms/AccelEmitter.cpp

```diff
@@ -213,11 +215,10 @@ ArrayAttr MfmaEmitter::computeOutputTransforms(PatternRewriter &b, Location loc,
 
   int64_t retNumElements = accVectorType.getNumElements();
   int64_t numElements = retNumElements * mRepeats * nRepeats * nResultVectors;
-  TopDownTMBuilder splitMemoryCoords(b, {"bid", "tid", "item"},
-                                     {gridSize, blockSize, numElements}, loc);
-
-  splitMemoryCoords.merge({"g", "m", "n"}, {0, 1, 2}, {"bid"},
-                          {gridSize / gStride, gStride / nBlocks, nBlocks});
+  TopDownTMBuilder splitMemoryCoords(
+      b, {"g_block", "m_block", "n_block", "tid", "item"},
+      {gridSize / gStride, mBlocks, nBlocks, blockSize, numElements}, loc);
```

Comment:
Can we not re-calculate grid coordinates here from gridSize ? 
The caller of computeOutputTransforms should already have bidGridLengths.